Lib/test/test_pyrepl/test_unix_console.py

```diff
@@ -303,3 +303,19 @@ def test_getheightwidth_with_invalid_environ(self, _os_write):
             self.assertIsInstance(console.getheightwidth(), tuple)
             os.environ = []
             self.assertIsInstance(console.getheightwidth(), tuple)
+
+    @unittest.skipUnless(sys.platform == "darwin", "requires OS X")
+    def test_is_mac_with_invalid_environ(self, _os_write):
+        # gh-128636 for macOS
+        console = UnixConsole(term="xterm")
+        with os_helper.EnvironmentVarGuard() as env:
+            is_mac = os.getenv("TERM_PROGRAM") == "Apple_Terminal"
+            self.assertEqual(console.is_mac, is_mac)
+            os.environ = []
+            console.prepare()  # Need to prepare before restore
+            try:
+                console.restore()  # This should not crash
+            except (KeyError, AttributeError, TypeError) as e:
+                self.fail(f"restore() failed with invalid environ: {e}")
+
+            self.assertEqual(console.is_mac, is_mac)
```

Comment:
```suggestion
    @unittest.skipUnless(sys.platform == "darwin", "requires OS X")
    def test_restore_with_invalid_environ_on_macOS(self, _os_write):
        # gh-128636 for macOS
        console = UnixConsole(term="xterm")
        with os_helper.EnvironmentVarGuard() as env:
            os.environ = []
            console.prepare()  # needed to call restore()
            console.restore()  # this should succeed
```

We don't need to test `is_mac`.